Dockerfile

```diff
@@ -1,13 +1,23 @@
-FROM golang:1.20-alpine as builder
+# syntax=docker/dockerfile:1
+# Build
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
+
+ARG TARGETOS TARGETARCH
+
+ENV GOOS=$TARGETOS \
+  GOARCH=$TARGETARCH \
+  CGO_ENABLED=0
```

Comment:
Is there a specific reason or benefit justifying setting the `CGO_ENABLED` as an `ENV` statement as opposed to prefix the `RUN` statement with it?

Reply:
Since I was adding already an `ENV`, I thought it would make sense to have all the environment variables together. Based on Docker's usage of cache per-line, ideally, changing the value of the `ENV` field would re-trigger the lower-layers.

This, in practical terms, does not matter much since adding an environment variable won't take much compute or storage. But stylistically, I think it looks way cleaner to have all the environment variables together, instead of leaving `CGO_ENABLED` by itself, alone, before the `go build` call.